OS9 Preserve Description after Defaulting Port

hakasapl opened the pull request:
These changes modify `conf_access_port.yaml` and `conf_trunk_port.yaml` to find and save the description of an interface before defaulting it, then reassigning it. This ensures description can be preserved for organization purposes.

naved001 commented:
Hi @radez! Can we have someone review this PR? Thanks!

cc: @larsks @tzumainn

naved001 commented on etc/ansible/roles/network-runner/providers/dellemc.os9.os9/conf_access_port.yaml:
```diff
@@ -1,10 +1,27 @@
 ---
+- name: "dellemc.os9: get running config for interface"
+  dellemc.os9.os9_command:
+    commands: "show running-config interface {{ port_name }}"
```
What if you filtered the output right at the switch?

`show running-config interface {{ port_name }} | grep description`
e.g.

```
OCT4-SW-TORS#show running-config interface tengigabitethernet 1/22 | grep description
 description OCT4-2 2nd
 ```
 
 and then the output can be directly applied

hakasapl replied:
That includes a space at the beginning of description (it's indented that way on the output), so there would need to be a bit more filtering regardless. I also think it's safer to use regex because the behavior of how to execute grep might change across versions, but the config output shouldn't change.

larsks reviewed (commented):
@hakasapl I think you should squash this down to a single commit (because the later commits are just fixing issues with the first commit), and update the commit message with some explanatory text (why are we doing this/what problem are we solving).

hakasapl committed:
OS9 port conf tasks now preserve description
Currently, conf_access_port and conf_trunk_port for OS9 clears the
description when defaulting the port. Descriptions should be
preserved for organizational purposes. Often times descriptions will
have information on the endpoint. This will keep that info by storing
description as a var then reapplying after the port is defaulted.

hakasapl commented:
> @hakasapl I think you should squash this down to a single commit (because the later commits are just fixing issues with the first commit), and update the commit message with some explanatory text (why are we doing this/what problem are we solving).

Sure, I have squashed the commits and added a description.
